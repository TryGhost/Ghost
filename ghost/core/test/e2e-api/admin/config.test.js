const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');
const {anyContentVersion, anyEtag, anyContentLength, stringMatching} = matchers;

/**
 * This is a snapshot test for the happy path of the config API
 * It does not test the full range of possible config values
 * as that should be tested in the unit tests for the public-config service
 */
describe('Config API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getAdminAPIAgent();
        await fixtureManager.init('users');
    });

    describe('As Unauthorized User', function () {
        it('Cannot fetch the config endpoint', async function () {
            await agent.get('/config/')
                .expectStatus(403);
        });
    });

    describe('As Owner', function () {
        before(async function () {
            await agent.loginAsOwner();
        });

        it('Can retrieve config and all expected properties', async function () {
            await agent
                .get('/config/')
                .expectStatus(200)
                .matchBodySnapshot({
                    config: {
                        database: stringMatching(/sqlite3|mysql|mysql2/),
                        environment: stringMatching(/^testing/),
                        version: stringMatching(/\d+\.\d+\.\d+/)
                    }
                })
                .matchHeaderSnapshot({
                    'content-version': anyContentVersion,
                    'content-length': anyContentLength, // Length can differ slightly based on the database, environment and version values
                    etag: anyEtag
                });
        });
    });
});
