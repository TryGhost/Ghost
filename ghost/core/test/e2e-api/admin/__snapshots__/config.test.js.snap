// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Config API As Owner Can retrieve config and all expected properties 1: [body] 1`] = `
Object {
  "config": Object {
    "clientExtensions": Object {},
    "database": StringMatching /sqlite3\\|mysql\\|mysql2/,
    "emailAnalytics": true,
    "enableDeveloperExperiments": false,
    "environment": StringMatching /\\^testing/,
    "labs": Object {
      "ActivityPub": true,
      "NestPlayground": true,
      "activitypub": true,
      "additionalPaymentMethods": true,
      "adminXDemo": true,
      "announcementBar": true,
      "audienceFeedback": true,
      "collections": true,
      "collectionsCard": true,
      "commentImprovements": true,
      "contentVisibility": true,
      "editorExcerpt": true,
      "emailCustomization": true,
      "i18n": true,
      "importMemberTier": true,
      "lexicalIndicators": true,
      "lexicalMultiplayer": true,
      "mailEvents": true,
      "members": true,
      "newEmailAddresses": true,
      "outboundLinkTagging": true,
      "stripeAutomaticTax": true,
      "themeErrorsNotification": true,
      "urlCache": true,
      "webmentions": true,
    },
    "mail": "",
    "mailgunIsConfigured": false,
    "signupForm": Object {
      "url": "https://cdn.jsdelivr.net/ghost/signup-form@~{version}/umd/signup-form.min.js",
      "version": "0.2",
    },
    "stripeDirect": false,
    "tenor": Object {
      "contentFilter": "off",
      "googleApiKey": null,
    },
    "useGravatar": false,
    "version": StringMatching /\\\\d\\+\\\\\\.\\\\d\\+\\\\\\.\\\\d\\+/,
  },
}
`;

exports[`Config API As Owner Can retrieve config and all expected properties 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "http://127.0.0.1:2369",
  "cache-control": "no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0",
  "content-length": StringMatching /\\\\d\\+/,
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Origin, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
